.github/workflows/ci-sglang-benchmark.yml

```diff
@@ -186,18 +176,65 @@ jobs:
 
       - name: Run SGLang Benchmark Tests
         run: |
-          pytest -v app_tests/benchmark_tests/llm/sglang_benchmarks/sglang_benchmark_test.py --port 30000 --log-cli-level=INFO --html=out/llm/sglang/index.html
+          pytest -v app_tests/benchmark_tests/llm/sglang_benchmarks/sglang_benchmark_test.py --port 30000 --log-cli-level=INFO --html=sglang_index.html --self-contained-html
 
       - name: Stop sglang-server
         run: docker stop sglang-server || true # Stop container if it's running
 
+      # Deleting image after run due to large disk space requirement (83 GB)
       - name: Cleanup SGLang Image
         run: docker image rm lmsysorg/sglang:v0.3.5.post1-rocm620
 
+      - name: Upload pytest report
+        uses: actions/upload-artifact@b4b15b8c7c6ac21ea08fcf65892d2ee8f75cf882
+        with:
+          name: sglang_benchmark
+          path: sglang_index.html
+
+  merge_and_upload_reports:
+    if: ${{ github.repository_owner == 'nod-ai' || github.event_name != 'schedule' }}
+    name: "Merge and upload benchmark reports"
+    needs: [benchmark_shortfin, benchmark_sglang]
```

Comment:
Is the duplicated `if` condition needed on this step here? I would expect the conditions in the `needs` steps to make this be skipped

https://stackoverflow.com/questions/65844033/conditional-needs-in-github-actions

One wrinkle to consider if how you want to handle one of the jobs succeeding and the other failing. Should that upload the results for the job that succeeded, or should it stop the combine step and the uploading?

Reply:
Duplicated `ifs` have been removed. 

That's a good point, really I just didn't want them to run at the same time. They're pretty resource heavy and didn't want that to cause the benchmark numbers to be off. However, if one fails, the other should still run. Updated test to handle that

Reply:
I would only use the setup you have if the workflows logically depend on one another:

```yml
  benchmark_sglang:
    name: "SGLang Serving Benchmark With SGLang"
    needs: benchmark_shortfin
    if: always()
```

Resource isolation and benchmark stability is the responsibility of the runner control plane and runner machines/images/VMs, not a single workflow. You have no control here over what other jobs might be running on the same physical machine. All your workflow should specify is what runner label it wants. There could be multiple physical machines with that label, or multiple runners on the same physical machine. Either way, the runners shouldn't be competing for resources.

Reply:
Gotcha, yeah that makes sense. I removed the dependency on `benchmark_shortfin` in `benchmark_sglang`. I then added an `if` statement for `merge_and_upload_reports` that waits for both jobs to finish, but still runs if either job completed successfully